test/client.js

```diff
@@ -0,0 +1,158 @@
+'use strict';
+
+const metatests = require('metatests');
+const { Metacom } = require('metacom/lib/client');
+
+let token = '';
+
+const runTests = (client) => {
+  metatests.testAsync('system/introspect', async (test) => {
+    const introspect = await client.socketCall('system')('introspect')([
+      'auth',
+      'console',
+      'example',
+      'files',
+      'test',
+    ]);
+    test.strictSame(introspect?.auth?.restore?.[0], 'token');
+    test.end();
+  });
+
+  metatests.testAsync('example/add', async (test) => {
+    const add = await client.socketCall('example')('add')({ a: 1, b: 2 });
+    test.strictSame(add, 3);
+    test.end();
+  });
+
+  metatests.testAsync('example/citiesByCountry', async (test) => {
+    const cities = await client.socketCall('example')('citiesByCountry')({
+      countryId: 1,
+    });
+    test.strictEqual(cities?.result, 'success');
+    test.strictEqual(Array.isArray(cities?.data), true);
+    test.end();
+  });
+
+  metatests.testAsync('example/customError', async (test) => {
+    try {
+      await client.socketCall('example')('customError')();
```

Comment:
```suggestion
      await client.socketCall('example')('customError')();
      test.fail('Shoul never reach here')
```

Use this patter for all throwing cases, to ensure that an error is always thrown.